packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -186,13 +199,36 @@ function NavigationMenuItemEdit( {
 					'is-selected': isSelected,
 				} ) }
 			>
-				<RichText
-					className="wp-block-navigation-menu-item__content"
-					value={ label }
-					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
-					placeholder={ __( 'Add item…' ) }
-					withoutInteractiveFormatting
-				/>
+				<KeyboardShortcuts
+					shortcuts={ {
+						[ rawShortcut.primary( 'k' ) ]: openLinkUI(),
+					} }
+				>
+					<RichText
+						className="wp-block-navigation-menu-item__content"
+						value={ label }
+						onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
+						placeholder={ __( 'Add item…' ) }
+						withoutInteractiveFormatting
+					/>
+				</KeyboardShortcuts>
+
+				{ isLinkOpen &&
+					<LinkControl
+						className="wp-block-navigation-menu-item__inline-link-input"
+						onKeyDown={ handleLinkControlOnKeyDown }
+						onKeyPress={ ( event ) => event.stopPropagation() }
+						currentLink={ link }
+						onLinkChange={ updateLink }
```

Comment:
Why not `onChange` for consistency with other controls?

Reply:
Thanks for the background — I think we should refine the exposed API of the component a bit, but should not be blocking this PR.